cmd/vsphere-cloud-controller-manager/main.go

```diff
@@ -56,38 +72,83 @@ func main() {
 	if err != nil {
 		klog.Fatalf("unable to initialize command options: %v", err)
 	}
-	c, err := s.Config([]string{}, app.ControllersDisabledByDefault.List())
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
 
-	// initialize cloud provider with the cloud provider name and config file provided
-	cloud, err := cloudprovider.InitCloudProvider(vsphere.ProviderName, c.ComponentConfig.KubeCloudShared.CloudProvider.CloudConfigFile)
-	if err != nil {
-		klog.Fatalf("Cloud provider could not be initialized: %v", err)
-	}
-	if cloud == nil {
-		klog.Fatalf("cloud provider is nil")
-	}
+	var controllerInitializers map[string]app.InitFunc
+	command := &cobra.Command{
+		Use:  "vsphere-cloud-controller-manager",
+		Long: `vsphere-cloud-controller-manager manages vSphere cloud resources for a Kubernetes cluster.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			verflag.PrintAndExitIfRequested()
+			cliflag.PrintFlags(cmd.Flags())
 
-	if !cloud.HasClusterID() {
-		if c.ComponentConfig.KubeCloudShared.AllowUntaggedCloud {
-			klog.Warning("detected a cluster without a ClusterID.  A ClusterID will be required in the future.  Please tag your cluster to avoid any future issues")
-		} else {
-			klog.Fatalf("no ClusterID found.  A ClusterID is required for the cloud provider to function properly.  This check can be bypassed by setting the allow-untagged-cloud option")
-		}
-	}
+			c, err := s.Config(KnownControllers(), app.ControllersDisabledByDefault.List())
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%v\n", err)
+				os.Exit(1)
+			}
+
+			klog.Infof("vsphere-cloud-controller-manager version: %s", version)
+
+			// initialize cloud provider with the cloud provider name and config file provided
+			cloud, err := cloudprovider.InitCloudProvider(vsphere.ProviderName, c.ComponentConfig.KubeCloudShared.CloudProvider.CloudConfigFile)
```

Comment:
my understanding is there is a flag or field in the config file that decides if `vsphere.ProviderName` or `vsphereparavirtual.ProviderName` should be used https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/cloudprovider/vsphereparavirtual/cloud.go#L37

Reply:
yes that's correct 
There will be a follow-up PR to enable paravirtual cloud provider